gtk2 clang build nastiness

```
C:/Msys2/clang32/include/glib-2.0/gobject/gtype.h:643:145: note: expanded from macro 'G_TYPE_INSTANCE_GET_PRIVATE'
#define G_TYPE_INSTANCE_GET_PRIVATE(instance, g_type, c_type)   ((c_type*) g_type_instance_get_private ((GTypeInstance*) (instance), (g_type))) GLIB_DEPRECATED_MACRO_IN_2_58_FOR(G_ADD_PRIVATE)
                                                                                                                                                ^
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: C:\Msys2\clang32\bin\i686-w64-mingw32-gcc.exe -DHAVE_CONFIG_H -I. -I../../gtk+-2.24.33/gtk -I.. -DGTK_VERSION="2.24.33" -DGTK_BINARY_VERSION="2.10.0" -DGTK_COMPILATION -DGTK_DISABLE_DEPRECATED -DGTK_FILE_SYSTEM_ENABLE_UNSUPPORTED -DGTK_PRINT_BACKEND_ENABLE_UNSUPPORTED -DGTK_LIBDIR="C:/Msys2/clang32/lib" -DGTK_DATADIR="C:/Msys2/clang32/share" -DGTK_DATA_PREFIX="C:/Msys2/clang32" -DGTK_SYSCONFDIR="C:/Msys2/clang32/etc" -DGTK_HOST="i686-w64-mingw32" #0 0x75e56ea3 (C:\Windows\System32\ucrtbase.dll+0x26ea3)
#1 0x75e592a0 (C:\Windows\System32\ucrtbase.dll+0x292a0)
#2 0x75e59152 (C:\Windows\System32\ucrtbase.dll+0x29152)
#3 0x5d909640 _ZN4llvm14raw_fd_ostream10write_implEPKcj (C:\Msys2\clang32\bin\libLLVM.dll+0x109640)
#4 0x5d8c0f9f _ZN4llvm21PrettyStackTraceEntryD2Ev (C:\Msys2\clang32\bin\libLLVM.dll+0xc0f9f)
#5 0x5d92426b _ZN4llvm3sys17RunSignalHandlersEv (C:\Msys2\clang32\bin\libLLVM.dll+0x12426b)
../libtool: linje 1779: 30833 Segmentation fault
```

any idea what might cause this ?

here is what I get right now:

```
libtool: link: if test DEF = "`/usr/bin/sed -n -e 's/^[  ]*//' -e '/^\(;.*\)*$/d' -e 's/^\(EXPORTS\|LIBRARY\)\([         ].*\)*$/DEF/p' -e q ../../gtk+-2.24.33/gdk/gdk.def`" ; then cp ../../gtk+-2.24.33/gdk/gdk.def .libs/libgdk-win32-2.0-0.dll.def; else echo EXPORTS > .libs/libgdk-win32-2.0-0.dll.def; cat ../../gtk+-2.24.33/gdk/gdk.def >> .libs/libgdk-win32-2.0-0.dll.def; fi
libtool: link:  clang -shared .libs/libgdk-win32-2.0-0.dll.def  .libs/gdk.o .libs/gdkapplaunchcontext.o .libs/gdkcairo.o .libs/gdkcolor.o .libs/gdkcursor.o .libs/gdkdisplay.o .libs/gdkdisplaymanager.o .libs/gdkdnd.o .libs/gdkdraw.o .libs/gdkevents.o .libs/gdkfont.o .libs/gdkgc.o .libs/gdkglobals.o .libs/gdkimage.o .libs/gdkkeys.o .libs/gdkkeyuni.o .libs/gdkoffscreenwindow.o .libs/gdkpango.o .libs/gdkpixbuf-drawable.o .libs/gdkpixbuf-render.o .libs/gdkpixmap.o .libs/gdkpolyreg-generic.o .libs/gdkrectangle.o .libs/gdkregion-generic.o .libs/gdkrgb.o .libs/gdkscreen.o .libs/gdkselection.o .libs/gdkvisual.o .libs/gdkwindow.o .libs/gdkwindowimpl.o .libs/gdkenumtypes.o .libs/gdkmarshalers.o .libs/gdkkeynames.o  -Wl,--whole-archive win32/.libs/libgdk-win32.a -Wl,--no-whole-archive  -limm32 -lshell32 -lole32 -LC:/msys64/clang64/lib -lpangowin32-1.0 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -lgio-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -lgdi32 -lmsimg32 -lcairo -lintl  -march=x86-64 -mtune=generic -O2 -mms-bitfields -Wl,win32/rc/gdk-win32-res.o -Wl,--dynamicbase -Wl,-luuid   -o .libs/libgdk-win32-2.0-0.dll -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/libgdk-win32-2.0.dll.a
lld-link: error: <root>: undefined symbol: gdk_test_simulate_button
lld-link: error: <root>: undefined symbol: gdk_test_simulate_key
lld-link: error: <root>: undefined symbol: gdk_test_render_sync
lld-link: error: <root>: undefined symbol: gdk_keyval_convert_case
lld-link: error: <root>: undefined symbol: gdk_keyval_from_name
lld-link: error: <root>: undefined symbol: gdk_keyval_name
lld-link: error: undefined symbol: IA__gdk_display_open
>>> referenced by .libs/gdk.o:(gdk_display_open_default_libgtk_only)
>>> referenced by .libs/gdk.o:(gdk_display_open_default_libgtk_only)

lld-link: error: undefined symbol: IA__gdk_screen_make_display_name
>>> referenced by .libs/gdkapplaunchcontext.o:(gdk_app_launch_context_get_display)

lld-link: error: undefined symbol: IA__gdk_display_get_name
>>> referenced by .libs/gdkapplaunchcontext.o:(gdk_app_launch_context_get_display)

lld-link: error: undefined symbol: IA__gdk_colormap_get_type
>>> referenced by .libs/gdkcolor.o:(gdk_colormap_get_visual)
>>> referenced by .libs/gdkgc.o:(gdk_gc_set_colormap)
>>> referenced by .libs/gdkimage.o:(gdk_image_set_colormap)
>>> referenced 4 more times

lld-link: error: undefined symbol: IA__gdk_colormap_change
>>> referenced by .libs/gdkcolor.o:(gdk_colors_store)

lld-link: error: undefined symbol: IA__gdk_colormap_alloc_colors
>>> referenced by .libs/gdkcolor.o:(gdk_color_white)
>>> referenced by .libs/gdkcolor.o:(gdk_colormap_alloc_color)
>>> referenced by .libs/gdkcolor.o:(gdk_color_black)
>>> referenced 1 more times

lld-link: error: undefined symbol: IA__gdk_screen_get_system_colormap
>>> referenced by .libs/gdkcolor.o:(gdk_colormap_get_system)
>>> referenced by .libs/gdkoffscreenwindow.o:(_gdk_offscreen_window_new)
>>> referenced by .libs/gdkrgb.o:(gdk_rgb_create_info)
>>> referenced 3 more times

lld-link: error: undefined symbol: IA__gdk_cursor_new_for_display
>>> referenced by .libs/gdkcursor.o:(gdk_cursor_new)

lld-link: error: undefined symbol: IA__gdk_display_pointer_ungrab
>>> referenced by .libs/gdkdisplay.o:(gdk_pointer_ungrab)
>>> referenced by .libs/gdkwindow.o:(gdk_window_hide)

lld-link: error: undefined symbol: IA__gdk_display_keyboard_ungrab
>>> referenced by .libs/gdkdisplay.o:(gdk_keyboard_ungrab)
>>> referenced by .libs/gdkwindow.o:(gdk_window_hide)

lld-link: error: undefined symbol: IA__gdk_display_beep
>>> referenced by .libs/gdkdisplay.o:(gdk_beep)
>>> referenced by .libs/gdkwindow.o:(gdk_window_beep)

lld-link: error: undefined symbol: IA__gdk_event_send_client_message_for_display
>>> referenced by .libs/gdkdisplay.o:(gdk_event_send_client_message)

lld-link: error: undefined symbol: IA__gdk_screen_broadcast_client_message
>>> referenced by .libs/gdkdisplay.o:(gdk_event_send_clientmessage_toall)

lld-link: error: undefined symbol: IA__gdk_screen_get_display
>>> referenced by .libs/gdkdisplay.o:(singlehead_default_window_at_pointer)
>>> referenced by .libs/gdkdraw.o:(gdk_drawable_get_display)
>>> referenced by .libs/gdkimage.o:(_gdk_image_get_scratch)
>>> referenced 4 more times

lld-link: error: too many errors emitted, stopping now (use /errorlimit:0 to see all errors)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [Makefile:1307: libgdk-win32-2.0.la] Error 1
make[4]: Leaving directory '/home/user/M/mingw-w64-gtk2/src/build-x86_64-w64-mingw32/gdk'
make[3]: *** [Makefile:1456: all-recursive] Error 1
make[3]: Leaving directory '/home/user/M/mingw-w64-gtk2/src/build-x86_64-w64-mingw32/gdk'
make[2]: *** [Makefile:1230: all] Error 2
make[2]: Leaving directory '/home/user/M/mingw-w64-gtk2/src/build-x86_64-w64-mingw32/gdk'
make[1]: *** [Makefile:740: all-recursive] Error 1
make[1]: Leaving directory '/home/user/M/mingw-w64-gtk2/src/build-x86_64-w64-mingw32'
make: *** [Makefile:632: all] Error 2
==> ERROR: A failure occurred in build().
```

--- a/mingw-w64-gtk2/0024-gtk-update-icon-cache-rc-quote.patch
+++ b/mingw-w64-gtk2/0024-gtk-update-icon-cache-rc-quote.patch
@@ -1,0 +1,11 @@
+--- gtk+-2.24.33/gtk/Makefile.am.orig	2021-07-10 23:22:04.765107300 -0700
++++ gtk+-2.24.33/gtk/Makefile.am	2021-07-10 23:22:27.889752400 -0700
+@@ -1128,7 +1128,7 @@
+ 	 echo '</assembly>' ) >$@
+ 
+ $(GTK_UPDATE_ICON_CACHE_RC):
+-	(echo -e '#include <winbase.h>\nCREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST '$(GTK_UPDATE_ICON_CACHE_MANIFEST)) >$@
++	(echo -e '#include <winbase.h>\nCREATEPROCESS_MANIFEST_RESOURCE_ID RT_MANIFEST "'$(GTK_UPDATE_ICON_CACHE_MANIFEST)'"') >$@
+ 
+ $(GTK_UPDATE_ICON_CACHE_MANIFEST_OBJECT): $(GTK_UPDATE_ICON_CACHE_RC) $(GTK_UPDATE_ICON_CACHE_MANIFEST)
+ 	$(WINDRES) --input $< --output $@ --output-format=coff

--- a/mingw-w64-gtk2/PKGBUILD
+++ b/mingw-w64-gtk2/PKGBUILD
@@ -5,7 +5,7 @@ _realname=gtk2
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=2.24.33
-pkgrel=3
+pkgrel=4
 pkgdesc="GTK+ is a multi-platform toolkit (v2) (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
@@ -43,7 +43,8 @@ source=("https://download.gnome.org/sources/gtk+/${pkgver%.*}/gtk+-${pkgver}.tar
         0012-embed-manifest.all.patch
         0013_fix_mouse_events.patch
         0022-icontheme-win32-detect-SVG-files-by-extension.patch
-        0023-gdkkeys-win32.c-fix-initialisation-of-key_state-in-u.patch)
+        0023-gdkkeys-win32.c-fix-initialisation-of-key_state-in-u.patch
+        0024-gtk-update-icon-cache-rc-quote.patch)
 
 sha256sums=('ac2ac757f5942d318a311a54b0c80b5ef295f299c2a73c632f6bfb1ff49cc6da'
             'b77a427df55a14182c10ad7e683b4d662df2846fcd38df2aa8918159d6be3ae2'
@@ -57,7 +58,8 @@ sha256sums=('ac2ac757f5942d318a311a54b0c80b5ef295f299c2a73c632f6bfb1ff49cc6da'
             '4bc70de68084585dcf2b5a7e4abe32a789360fc2be97a9c94ba3b52dac89ad93'
             'f6d731acecf6e35e4ec9739a6d949d2d045c9396c4564283ee6ec935ec690e24'
             'b0f18b85d2fd47057c266887848be1f1c0b6171a1ee6d52a33df7af6e0abb8b0'
-            '68f9c7ea1a548a5d700aa4ad17bb961815ad59cc8d05fc1af156ef680da24a15')
+            '68f9c7ea1a548a5d700aa4ad17bb961815ad59cc8d05fc1af156ef680da24a15'
+            '9c6e2646b22c9dc1725595dbd5fedcaa71ea85ddb5611a1bccc130a00e9e7e30')
 
 prepare() {
   cd gtk+-${pkgver}
@@ -74,6 +76,9 @@ prepare() {
   
   # https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/3741
   patch -p1 -i ${srcdir}/0023-gdkkeys-win32.c-fix-initialisation-of-key_state-in-u.patch
+
+  # https://github.com/msys2/MINGW-packages/issues/9130
+  patch -p1 -i ${srcdir}/0024-gtk-update-icon-cache-rc-quote.patch
 
   autoreconf -fi
 }
@@ -94,6 +99,7 @@ build() {
     --enable-shared \
     --enable-introspection \
     --disable-glibtest \
+    --disable-visibility \
     --with-included-immodules=ime
 
   make

